packages/content-slider/src/components/content-slider/content-slider.component.ts

```diff
@@ -0,0 +1,280 @@
+import { TemplatesComponent } from "@ribajs/core";
+import { throttle } from "@ribajs/utils/src/control";
+import { hasChildNodesTrim } from "@ribajs/utils/src/dom";
+
+import template from "./content-slider.component.html";
+
+export interface SlideItem {
+  /** Template content is the item capture */
+  content: string;
+  imageSrc: string;
+  active: boolean;
+  index: number;
+}
+
+interface Scope {
+  items: SlideItem[];
+  currentIndex: number;
+  activeItem: SlideItem | null;
+  transform: string;
+  previous: ContentSliderComponent["previous"];
+  next: ContentSliderComponent["next"];
+  goTo: ContentSliderComponent["goTo"];
+  activeItemWidth: number;
+  inactiveItemWidth: number;
+  activeClass: string;
+  activeColumnClasses: string[];
+  inactiveColumnClasses: string[];
+}
+
+export class ContentSliderComponent extends TemplatesComponent {
+  public static tagName = "content-slider";
+
+  protected autobind = true;
+
+  protected templateAttributes = [
+    {
+      name: "image-src",
+      required: true,
+    },
+    {
+      name: "active",
+      required: false,
+    },
+    {
+      name: "index",
+      required: false,
+    },
+  ];
+
+  static get observedAttributes() {
+    return ["active-class", "active-column-classes", "inactive-column-classes"];
+  }
+
+  protected scope: Scope = {
+    items: [],
+    currentIndex: 0,
+    activeItem: null,
+    transform: "translateX(0)",
+    previous: this.previous,
+    next: this.next,
+    goTo: this.goTo,
+    activeItemWidth: 0,
+    inactiveItemWidth: 0,
+    activeClass: "active",
+    activeColumnClasses: [
+      "col-10",
+      "col-sm-10",
+      "col-md-6",
+      "col-lg-5",
+      "col-xl-4",
+    ],
+    inactiveColumnClasses: [
+      "col-6",
+      "col-sm-6",
+      "col-md-2",
+      "col-lg-2",
+      "col-xl-2",
+    ],
+  };
+
+  constructor(element?: HTMLElement) {
+    super(element);
+    this.debug("constructor", this);
+  }
+
+  protected connectedCallback() {
+    super.connectedCallback();
+    return this.init(ContentSliderComponent.observedAttributes);
+  }
+
+  protected initItems() {
+    const items = this.el.querySelectorAll(".item");
+    if (!items) {
+      throw new Error("No required items found!");
+    }
+
+    this.setActiveItem(0);
+
+    for (let i = 0; i < this.scope.items.length; i++) {
+      const item = items[i];
+      this.setInactiveClasses(item);
+    }
+
+    this.goTo(this.scope.currentIndex);
```

Comment:
`setActiveItem(0)` passiert sowieso in `this.goTo(this.scope.currentIndex)` und hat vorher auch keinen weiteren Effekt.

Gut wäre wohl, wenn man ein anderes Item als das erste zu Beginn als "aktiv" markieren kann (template mit Attribut "active", oder Attribut `<contentSlider "initialItem=index" ...`), und die Komponente dann hier `setActive(initialIndex)` setzt, welches Item als erstes aktiv sein soll.

Reply:
Danke, ich habe jetzt `currentIndex` allgemeiner zu `index` umbenannt und zu den observedAttributes hinzugefügt, damit kann der start-index nun über das attribute `index` selbst festgelegt werden. Außerdem kann der wert nun auch on-the-fly geändert werden, damit hat man nun z.B. auch die Möglichkeit durch ändern dieses Attributes das aktive Item zu wechseln.